exercise_004_extension_methods/README.md

```diff
@@ -4,10 +4,59 @@
 
 Extension methods can be used to add methods to types after they are defined.
 
-## Steps
+Lets say we need to add a method `square` to type `Int`. We used to achieve it 
+with the help of `implicit class`es in _Scala 2_
+
+```scala
+implicit class RichInt(val i: Int) extends AnyVal {
+  def square = i * i
+}
+```
+
+Now you can use it as follows
+
+```scala
+4.square
+// res1: Int = 16
+```
+
+With dotty `extension methods`, we can rewrite the above as
+
+```scala
+def (i: Int).square: Int = i * i
+
+3.square
+// val res0: Int = 9
+```
+
+More than one extension methods can be wrapped inside an `Extension Instance`.
```

Comment:
I would rephrase this as:

When a series of extension methods need to be defined on the same type,
encoding them one by one quickly becomes tedious. So-called
`Collective Extensions` can group the definitions together.